UnityProject/Assets/Scripts/Items/Others/Jetpack.cs

```diff
@@ -0,0 +1,68 @@
+using Objects.Atmospherics;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Items.Others
+{
+	/// <summary>
+	/// A jetpack that allows players to move freely in low gravity tiles (such as in space)
+	/// It does this by force pushing the player to the direction they're facing when a movement key is pressed.
+	/// </summary>
+	public class Jetpack : MonoBehaviour, IInteractable<InventoryApply>
+	{
+		[SerializeField] private float gasReleaseOnUse = 0.2f;
+		private Pickupable pickupable;
+		private GasContainer gasContainer;
+
+		private bool isOn;
+		private PlayerScript player;
+
+
+		private void Awake()
+		{
+			pickupable = GetComponent<Pickupable>();
+			gasContainer = GetComponent<GasContainer>();
+		}
+
+		/// <summary>
+		/// The jetpack is only usable inside the player's inventory slots so we check if it's
+		/// in his hand or equipment slot.
+		/// </summary>
+		private bool CheckForInventory()
+		{
+			//Is the jetpack inside a slot right now?
+			if (pickupable.ItemSlot == null) return false;
+			//Is this slot a player one? True if hand or equipment false if in a backpack or such.
+			if (pickupable.ItemSlot.Player == null) return false;
+			return true;
+		}
+
+		private void PushUpdate()
+		{
+			if (isOn == false || CheckForInventory() == false) return;
+			if (gasContainer.GasMix.Moles <= 0) return;
+			if (KeyboardInputManager.IsMovementPressed() || player.PlayerSync.IsMoving)//Is movement pressed?
```

Comment:
IsMoving just means if the character is moving, not if movement keys are pressed. 

You'll be moving if you are drifting in space and then would trigger this.

Reply:
any idea what I can use to check if movement keys have been pressed on the server?

Reply:
I don't think we send what you need currently as all the movement stuff will be client checked before, and so if they are in space floating then you wont try to tell the server to move. 

You'll need to see where the client side bit of the movement is (You'll have to follow the stuff in the keyboard manager) and add a new message if they are wearing a jetpack (or make it a generic message)

Reply:
Done